prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -404,6 +415,19 @@ ssize_t rxr_pkt_init_eager_msgrtm(struct rxr_ep *ep,
 	return 0;
 }
 
+ssize_t rxr_pkt_init_dc_eager_msgrtm(struct rxr_ep *ep,
+				     struct rxr_tx_entry *tx_entry,
+				     struct rxr_pkt_entry *pkt_entry)
+{
+	struct rxr_dc_eager_msgrtm_hdr *dc_eager_msgrtm_hdr;
+
+	rxr_pkt_init_rtm(ep, tx_entry, RXR_DC_EAGER_MSGRTM_PKT, 0, pkt_entry);
+	dc_eager_msgrtm_hdr = rxr_get_dc_eager_msgrtm_hdr(pkt_entry->pkt);
+	dc_eager_msgrtm_hdr->hdr.flags |= RXR_REQ_WAIT_RECEIPT;
```

Comment:
Not sure if I missed this earlier or if it changed. Is there a reason we're sending this flag over the wire? That seems wasteful.

Reply:
This flag is a way to notify receiver should post a receipt packet back to the sender upon receiving this packet. It is introduced for convenience to avoid branching and long list of conditions.

Reply:
Convenience isn't really applicable here, either it should be in the wire protocol or it shouldn't.

I'm not sure I understand where all the branching is going to be introduced. We already have a huge list of conditions in `rxr_pkt_handle_recv_completion`, why not refactor that and set the flag when the DC packet comes in?

Reply:
`RXR_REQ_WAIT_RECEIPT` flag is no longer used.